sys-utils/setpriv.c

```diff
@@ -459,6 +463,15 @@ static void parse_pdeathsig(struct privctx *opts, const char *str)
 	}
 }
 
+static void parse_ptracer(struct privctx *opts, const char *str)
+{
+	if (!strcmp(str, "any")) {
+		opts->ptracer = PR_SET_PTRACER_ANY;
+	} else {
+		opts->ptracer = strtol_or_err(str, _("failed to parse ptracer pid"));
```

Comment:
This allows negative values, 0, and too large values.
`strtopid_or_err()` prevents those.

Reply:
Updated, thanks!

One thing is that `prctl(PR_SET_PTRACER, 0)` is meaningful in that it disables a previously set ptracer. Should we support the ability to do that, maybe with `--ptracer none`?

Reply:
This sounds like a very fringe usecase, but given that it's trivial to support wh not.